src/scenic/simulators/robosuite/simulator.py

```diff
@@ -0,0 +1,566 @@
+        
+        objects.append(config)
+    
+    def _get_robot_arg(self, robots: List) -> Union[str, List[str]]:
+        """Get robot argument for environment creation."""
+        if not robots:
+            raise ValueError("At least one robot is required")
+        return ([r['type'] for r in robots] if len(robots) > 1 
+                else robots[0]['type'])
+    
+    def _detect_controller_type(self, robot_configs: List):
+        """Detect controller type from first robot."""
+        if robot_configs and self.robosuite_env.robots:
+            first_robot = self.robosuite_env.robots[0]
+            if hasattr(first_robot, 'controller'):
+                self.controller_type = type(first_robot.controller).__name__
+                if hasattr(first_robot.controller, 'control_dim'):
+                    self.action_dim = first_robot.controller.control_dim
+    
+    def _setup_body_mapping(self):
+        """Map environment objects to MuJoCo body IDs."""
+        for obj in self.objects:
+            if obj in self.object_name_map:
+                self._map_object_body(obj)
+    
+    def _map_object_body(self, obj):
+        """Map single object to body ID."""
+        obj_name = self.object_name_map.get(obj)
+        if not obj_name:
+            return
+            
+        for suffix in BODY_SUFFIXES:
+            body_name = f"{obj_name}{suffix}"
+            try:
+                body_id = mujoco.mj_name2id(
+                    self.model, mujoco.mjtObj.mjOBJ_BODY, body_name
+                )
+                if body_id != -1:
+                    self.body_id_map[obj_name] = body_id
+                    self.prev_positions[obj_name] = self.data.xpos[body_id].copy()
+                    break
+            except:
+                continue
+    
+    def createObjectInSimulator(self, obj):
+        """Required by Scenic's Simulator interface."""
+        # Objects are created during setup in ScenicManipulationEnv
+        pass
+    
+    def executeActions(self, allActions: Dict[Any, List]) -> None:
+        """Execute actions by calling their applyTo methods."""
+        super().executeActions(allActions)
+        
+        self.pending_robot_action = None
+        
+        for agent in self.agents:
+            if agent in allActions and allActions[agent]:
+                for action in allActions[agent]:
+                    if action and hasattr(action, 'applyTo'):
+                        action.applyTo(agent, self)
+                        break
+    
+    def step(self):
+        """Step the simulation forward one timestep."""
+        if hasattr(self, '_done') and self._done:
+            return
+        
+        # Store previous positions
+        for name, body_id in self.body_id_map.items():
+            self.prev_positions[name] = self.data.xpos[body_id].copy()
+        
+        action = (self.pending_robot_action if self.pending_robot_action is not None 
```

Comment:
This seems to assume there's only a single robot. We should support multiple robots taking actions at the same time.